### PR TITLE
Added the Rating model

### LINKED ISSUE #55: Adding the Rating model
- Adding the Rating model and related functions to the Resume model
- Adding tests to the Rating model and the functions.

### PR DESCRIPTION
**Pull request by AlonShakaroff**
- Added the rating table and functions
- Added functions to resume that refer to the ratings of the function
- Added tests to the Rating, and to the Resume functions.

fix #55 
depends on #50

### CODE REVIEW

**Review by OmerCS8 — DISMISSED** [review]
LGTM 🥇

**Review by TomerNewmanPrograms — APPROVED** [review]
lgtm

**Review by yulisuliman — APPROVED** [review]

**Review by MatanP12 — APPROVED** [review]
LGTM

**Review by OmerCS8 — APPROVED** [review]
LGTM

File: posts/tests.py [hunk 1]
```diff
@@ -1,3 +1,201 @@
...
+
+    # Check if the Resume can be referred as a post.
+    def test_resume_as_post(self, new_resume):
+        assert len(Post.objects.all()) == 0
+        new_resume.author.save()
+        new_resume.save()
+        assert isinstance(new_resume, Post)
+        assert len(Post.objects.all()) == 1
+        post = Post.objects.all().first()
+        assert hasattr(post, 'resume') is True
```
Comment (MatanP12) [inline posts/tests.py:84]: change here as well

File: posts/tests.py [hunk 2]
```diff
@@ -1,3 +1,201 @@
...
+        assert hasattr(post, 'poll') is False
```
Comment (MatanP12) [inline posts/tests.py:85]: same

**Review by MatanP12 — CHANGES_REQUESTED** [review]

File: posts/models.py [hunk 3]
```diff
@@ -1,13 +1,91 @@
...
+        for rating in ratings:
+            rating_sum += rating.grammar_rating
+        return rating_sum / len(self.rating_set.all())
+
+    def get_conciseness_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.conciseness_rating
+        return rating_sum / len(self.rating_set.all())
```
Comment (natifridman) [inline posts/models.py:60]: I think you can use these methods as [class method](https://stackabuse.com/pythons-classmethod-and-staticmethod-explained/), so you can use [cls.objects.filter](https://github.com/redhat-beyond/falcon/blob/c719a8365b2f146224f763478d978acad5c39c02/tasks/models.py#L37)

Comment (AlonShakaroff) [reply posts/models.py:60]: Ok, I'll check that out, thanks!

Comment (AlonShakaroff) [reply posts/models.py:60]: If I use these classes as class methods I can't use the object it self to get it's rating_set, so I don't see how am I supposed to get the rating average with the method as class method? @natifridman

Comment (natifridman) [reply posts/models.py:60]: I thought it was in the Rating model, you can ignore this comment if you keep it in the Resume model

**Review by MatanP12 — CHANGES_REQUESTED** [review]
Please delete the .idea folder from the PR

**Comment by AlonShakaroff** [comment]
> Please delete the .idea folder from the PR

Oops, my bad, fixed it 👍

**Review by OmerCS8 — DISMISSED** [review]
LGTM

**Review by TomerNewmanPrograms — APPROVED** [review]

**Review by TomerNewmanPrograms — APPROVED** [review]

**Review by OmerCS8 — APPROVED** [review]

File: posts/models.py [hunk 4]
```diff
@@ -21,4 +23,70 @@ class Resume(Post):
...
```
Comment (natifridman) [inline posts/models.py:61]: same

File: posts/models.py [hunk 5]
```diff
@@ -21,4 +23,70 @@ class Resume(Post):
...
     def __str__(self):
-        return f"ResuMe {self.post_id} by {self.author}"
+        return f"Resume {self.post_id} by {self.author}"
+
+    def get_average_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        for rating in ratings:
+            rating_sum += rating.get_rating_average()
+        return rating_sum / len(self.rating_set.all())
```
[the remark this answers is not in the document]
Comment (AlonShakaroff) [reply posts/models.py]: Yeah, you're right, changed that in all the methods 👍 
@natifridman

**Review by OmerCS8 — DISMISSED** [review]
🚀

**Review by MatanP12 — APPROVED** [review]
🔥

**Review by TomerNewmanPrograms — DISMISSED** [review]

**Comment by natifridman** [comment]
@AlonShakaroff Maybe we can add some tests for the average methods?

**Comment by AlonShakaroff** [comment]
> @AlonShakaroff Maybe we can add some tests for the average methods?

I wrote them already and didn't upload them by mistake, I'll upload them soon.

**Comment by AlonShakaroff** [comment]
> @AlonShakaroff Maybe we can add some tests for the average methods?

Added the tests for the methods, and also fixed potential division by zero bugs.
@natifridman

**Review by OmerCS8 — APPROVED** [review]

**Review by yulisuliman — APPROVED** [review]
Looks good!

File: posts/models.py [hunk 6]
```diff
@@ -21,4 +23,92 @@ class Resume(Post):
...
 
     def __str__(self):
-        return f"ResuMe {self.post_id} by {self.author}"
+        return f"Resume {self.post_id} by {self.author}"
+
+    def get_average_rating(self):
+        ratings = self.rating_set.all()
+        rating_sum = 0
+        ratings_len = len(ratings)
+
```
Comment (natifridman) [inline posts/models.py:32]: Please remove the [blank line](https://peps.python.org/pep-0008/#blank-lines)

File: posts/models.py [hunk 7]
```diff
@@ -21,4 +23,92 @@ class Resume(Post):
...
+        for rating in ratings:
+            rating_sum += rating.get_rating_average()
+
```
Comment (natifridman) [inline posts/models.py:35]: Please remove the [blank line](https://peps.python.org/pep-0008/#blank-lines)

File: posts/tests.py [hunk 8]
```diff
@@ -75,6 +118,24 @@ def test_resume_deletion_after_author_deletion(self, persist_resume):
         assert persist_resume not in Resume.objects.all()
         assert user not in User.objects.all()
 
+    # Check if the Resume average methods work properly.
+    def test_resume_average_functions(self, persist_resume):
+        current_grade = 5
+        current_username = "test_rater"
+
```
Comment (natifridman) [inline posts/tests.py:125]: Please remove the [blank line](https://peps.python.org/pep-0008/#blank-lines)

File: posts/tests.py [hunk 9]
```diff
@@ -75,6 +118,24 @@ def test_resume_deletion_after_author_deletion(self, persist_resume):
...
+        for i in range(1, 4):
+            current_username += str(i)
+            current_email = current_username + "@gmail.com"
+            current_rater = new_user_with_name_and_email(current_username, current_email)
+            new_rating_with_input(resume=persist_resume, author=current_rater, grade=current_grade)
+            current_grade -= 2
+
```
Comment (natifridman) [inline posts/tests.py:132]: Please remove the [blank line](https://peps.python.org/pep-0008/#blank-lines)

**Review by natifridman — CHANGES_REQUESTED** [review]
Please follow [pep8 for blank line style](https://peps.python.org/pep-0008/#blank-lines)

**Comment by AlonShakaroff** [comment]
> Please follow [pep8 for blank line style](https://peps.python.org/pep-0008/#blank-lines)

Removed the blank lines. @natifridman

**Review by yulisuliman — APPROVED** [review]

**Review by MatanP12 — APPROVED** [review]
LGTM

**Review by natifridman — APPROVED** [review]

**Pull request merged**